### PR TITLE
Refactor/landgrif 818 actual vs scenario impact

### PR DESCRIPTION
**Pull request by yulia-bel**
### General description

Update service that compares actual impact with the selected scenario impact:
- Services for normal impact table and actual-data-vs-scenario comparison impact table are now separated
- Calculations methods updated in accordance with recent changes

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Provide minimal instructions on how to test this PR._

- Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging

- [ ] Branch name / PR includes the related Jira ticket Id.
- [ ] Tests to check core implementation / bug fix added.
- [ ] All checks in Continuous Integration workflow pass.
- [ ] Feature functionally tested by reviewer(s).
- [ ] Code reviewed by reviewer(s).
- [ ] Documentation updated (README, CHANGELOG...) (if required)

### CODE REVIEW

File: api/src/modules/impact/impact.controller.ts [hunk 1]
```diff
@@ -44,6 +49,26 @@ export class ImpactController {
...
 
+  @ApiOperation({
+    description:
+      'Get data for comapring Actual data with Scenario in form of Impact Table',
+  })
+  @ApiOkResponse({
+    type: PaginatedImpactTable,
+  })
+  @JSONAPIPaginationQueryParams()
+  @Get('actual-vs-scenario')
```
Comment (yulia-bel) [inline api/src/modules/impact/impact.controller.ts:60]: compare/scenario/vs/actual

File: api/src/modules/impact/actual-vs-scenario.service.ts [hunk 2]
```diff
@@ -0,0 +1,627 @@
...
+        entities,
+        fetchSpecification,
+      );
+
+    this.updateGroupByCriteriaFromEntityTree(
+      actualVsScenarioImpactTableDto,
+      paginatedEntities.entities,
+    );
+
+    const dataForActualVsScenarioImpactTable: ActualVsScenarioImpactTableData[] =
```
Comment (yulia-bel) [inline api/src/modules/impact/actual-vs-scenario.service.ts:84]: here type should be ImpactTableData[]

File: api/src/modules/impact/actual-vs-scenario.service.ts [hunk 3]
```diff
@@ -0,0 +1,627 @@
...
+      await this.getDataForActualVsScenarioImpactTable(
+        actualVsScenarioImpactTableDto,
+        paginatedEntities.entities,
+      );
+
+    const processedDataForComparison: ActualVsScenarioImpactTableData[] =
```
Comment (yulia-bel) [inline api/src/modules/impact/actual-vs-scenario.service.ts:90]: and here this method transforms ImpactTableData[] to ActualVsScenarioImpactTableData[]

File: api/src/modules/impact/actual-vs-scenario.service.ts [hunk 4]
```diff
@@ -0,0 +1,627 @@
...
+        totalPages: Math.ceil(totalItems / pageSize),
+        totalItems,
+        size: pageSize,
+        page,
+      }),
+    };
+  }
+
+  private static processDataForComparison(
+    dataForActualVsScenarioImpactTable: ActualVsScenarioImpactTableData[],
```
Comment (yulia-bel) [inline api/src/modules/impact/actual-vs-scenario.service.ts:555]: here data me receive should be of type ImpactTableData[], and the return result of type ActualVsScenarioImpactTableData[]

File: api/src/modules/impact/actual-vs-scenario.service.ts [hunk 5]
```diff
@@ -0,0 +1,627 @@
...
+    entities: ImpactTableEntityType[],
+  ): Promise<ActualVsScenarioImpactTableData[]> {
+    return entities.length > 0
+      ? this.sourcingRecordService.getDataForActualVsScebarioImpactTable(
+          actualVsScenarioImpactTableDto,
+        )
+      : Promise.resolve([]);
+  }
+
+  private buildImpactTable(
```
Comment (yulia-bel) [inline api/src/modules/impact/actual-vs-scenario.service.ts:232]: buildActualVsScenarioImpactTable

File: api/src/modules/impact/actual-vs-scenario.service.ts [hunk 6]
```diff
@@ -0,0 +1,627 @@
...
+              )
+                ? currentValue.values[indexOfYear].value
+                : 0;
+            return accumulator;
+          },
+          0,
+        );
+
+        let totalScenarioSumByYear: number | null = null;
+        if (queryDto.scenarioId) {
```
Comment (yulia-bel) [inline api/src/modules/impact/actual-vs-scenario.service.ts:331]: remove if

**Review by alexeh — APPROVED** [review]

**Pull request merged**